tasmota/tasmota_xsns_sensor/xsns_39_max31855.ino

```diff
@@ -114,54 +124,67 @@ float MAX31855_GetReferenceTemperature(int32_t RawData) {
-  } else {
-    int32_t RawData = MAX31855_ShiftIn(32);
-    uint8_t probeerror = RawData & 0x7;
-
-    MAX31855_Result.ErrorCode = probeerror;
-    MAX31855_Result.ReferenceTemperature = MAX31855_GetReferenceTemperature(RawData);
-    if (probeerror) {
-      MAX31855_Result.ProbeTemperature = NAN;    // Return NaN if MAX31855 reports an error
-    } else {
-      MAX31855_Result.ProbeTemperature = MAX31855_GetProbeTemperature(RawData);
+  for (uint32_t i = 0; i < MAX_MAX31855S; i++) {
+    if (max31855_pins_used & (1 << i)) {
+      if (Settings->flag4.max6675) {                  // SetOption94 - Implement simpler MAX6675 protocol instead of MAX31855
+        int32_t RawData = MAX31855_ShiftIn(16, i);
+        int32_t temp = (RawData >> 3) & ((1 << 12) - 1);
+
+        /* Occasionally the sensor returns 0xfff, consider it an error */
+        if (temp == ((1 << 12) - 1)) { return; }
+
+        MAX31855_Result[i].ErrorCode = 0;
+        MAX31855_Result[i].ReferenceTemperature = NAN;
+        MAX31855_Result[i].ProbeTemperature = ConvertTemp(0.25f * temp);
+      } else {
+        int32_t RawData = MAX31855_ShiftIn(32, i);
+        uint8_t probeerror = RawData & 0x7;
+
+        MAX31855_Result[i].ErrorCode = probeerror;
+        MAX31855_Result[i].ReferenceTemperature = MAX31855_GetReferenceTemperature(RawData);
+        if (probeerror) {
+          MAX31855_Result[i].ProbeTemperature = NAN;    // Return NaN if MAX31855 reports an error
+        } else {
+          MAX31855_Result[i].ProbeTemperature = MAX31855_GetProbeTemperature(RawData);
+        }
+      }
     }
   }
 }
 
 void MAX31855_Show(bool Json) {
-  char sensor_name[10];
-  GetTextIndexed(sensor_name, sizeof(sensor_name), Settings->flag4.max6675, kMax31855Types);
-
-  if (Json) {
-    ResponseAppend_P(PSTR(",\"%s\":{\"" D_JSON_TEMPERATURE "\":%*_f,\"" D_JSON_REFERENCETEMPERATURE "\":%*_f,\"" D_JSON_ERROR "\":%d}"), \
-      sensor_name,
-      Settings->flag2.temperature_resolution, &MAX31855_Result.ProbeTemperature,
-      Settings->flag2.temperature_resolution, &MAX31855_Result.ReferenceTemperature,
-      MAX31855_Result.ErrorCode);
+  char sensor_name_text[10];
+  char sensor_name[12];
+  GetTextIndexed(sensor_name_text, sizeof(sensor_name_text), Settings->flag4.max6675, kMax31855Types);
+  sprintf(sensor_name, "%s",sensor_name_text);
+  for (uint32_t i = 0; i < MAX_MAX31855S; i++) {
+    if (max31855_pins_used & (1 << i)) {
+      if (max31855_count > 1) {
+        sprintf(sensor_name, "%s%c%d",sensor_name_text, IndexSeparator(), i);
+      }
+
+      if (Json) {
+        ResponseAppend_P(PSTR(",\"%s\":{\"" D_JSON_TEMPERATURE "\":%*_f,\"" D_JSON_REFERENCETEMPERATURE "\":%*_f,\"" D_JSON_ERROR "\":%d}"), \
+          sensor_name,
+          Settings->flag2.temperature_resolution, &MAX31855_Result[i].ProbeTemperature,
+          Settings->flag2.temperature_resolution, &MAX31855_Result[i].ReferenceTemperature,
+          MAX31855_Result[i].ErrorCode);
 #ifdef USE_DOMOTICZ
```

Comment:
Do not update both Domoticz and KNX with different sensors. See xsns_47_max31865 how to solve